### ISSUE TITLE
Not able to read

### ISSUE THREAD

**Issue by xafilox** [dfdx/Kafka.jl#1]
Hi. I'm trying to execute the example of the readme with Kafka 2.1.0 Julia 0.5.0 but it is not working. It writes correctly, but while reading it keeps waiting and doesn't return. The topic was completely empty for the example.

Doing some debugging with prints I see that in the handle_response method in client.jl it receives multiple responses with the following values for the correlation_id
> handle_response - Header {"correlation_id":1}
> handle_response - Header {"correlation_id":614764}
> handle_response - Header {"correlation_id":16777216}
> handle_response - Header {"correlation_id":0}
> handle_response - Header {"correlation_id":23808}
> handle_response - Header {"correlation_id":0}
> handle_response - Header {"correlation_id":67702016}
> handle_response - Header {"correlation_id":19988480}
> handle_response - Header {"correlation_id":1701576704}
> handle_response - Header {"correlation_id":65536}
> handle_response - Header {"correlation_id":-1211957248}
> handle_response - Header {"correlation_id":838860800}
> handle_response - Header {"correlation_id":1677721600}
> handle_response - Header {"correlation_id":33554432}
> handle_response - Header {"correlation_id":100663296}
> handle_response - Header {"correlation_id":4}

So I suppose it is for some reason trying to read with a different message format than the one Kafka is using.
Thanks.

**Comment by xafilox** [comment]
I have inspected the communication with Wireshark and I think I have found the problem. The information of the fetch response is divided (for some reason) into two TCP frames that contain the following data (hex and ascii representation):
```
0000   00 00 00 86 00 00 00 01 **00 00 00 01**              ............

0000   00 09 61 6c 65 6a 61 6e 64 72 6f 00 00 00 01 00  ..alejandro.....
0010   00 00 00 00 00 00 00 00 00 00 00 00 03 00 00 00  ................
0020   5d 00 00 00 00 00 00 00 00 00 00 00 13 f8 04 09  ]...............
0030   0d 00 00 00 00 00 01 31 00 00 00 04 66 65 65 6c  .......1....feel
0040   00 00 00 00 00 00 00 01 00 00 00 13 80 96 b7 c3  ................
0050   00 00 00 00 00 01 32 00 00 00 04 67 6f 6f 64 00  ......2....good.
0060   00 00 00 00 00 00 02 00 00 00 13 4c 4c 59 06 00  ...........LLY..
0070   00 00 00 00 01 33 00 00 00 04 69 6e 63 2e        .....3....inc.
```

The, basically I have added println to all the methods in io.jl as well as handle_response method in client.jl. And I obtain the following trace for the fetch response:

> Reading Int32 with value 134 (86)
> io - recv_header - size 134
> reading object of Kafka.ResponseHeader
> Reading Int32 with value 1 (1)
> handle_response - Header {"correlation_id":1}
> handle_response - T Kafka.FetchResponse_v0
> reading object of Kafka.FetchResponse_v0
> **Reading Int32 with value 614764 (9616c)**
> reading vector of Kafka.FetchResponseTopicData with size 614764 (9616c)
> reading object of Kafka.FetchResponseTopicData
> Reading Int16 with value 25962 (656a)
> Reading string of length 25962
> ...

So when it is reading what is supposed to be the size of the vector of type FetchResponseTopicData it is reading 0009616c instead of 00000001 (marked between asterisks in the network dump). So it seems that it is not receiving the last 4 bytes of the first TCP frame... Any idea of what is happening?

**Comment by dfdx** [comment]
Thanks for reporting, I'll take a look at it as soon as possible. Could you please verify Kafka version? Currently, the latest version of Kafka is 0.10.1, so probably you refer to Hadoop / Confluent / another platform version.

**Comment by xafilox** [comment]
Thanks for taking the time to look at it. Yes, sorry, the correct version is 0.10.0

**Comment by dfdx** [comment]
I have updated the master to fix deprecation warnings for Julia 0.5, please update. 

However, I wasn't able to reproduce your error on kafka_2.11-0.10.0.1. Do you copy-paste example from README exactly or have any changes? If so, could you please provide the code you use? 

If the code is the same and it still doesn't work on the latest master, could you please specify exact Kafka version you use?

**Comment by xafilox** [comment]
I have updated to the last changes you made in master and now it seems to be working. Thanks a lot for your help!
